src/components/edit-in-place/index.js

```diff
@@ -26,15 +24,19 @@ const EditInPlace = forwardRef(
       []
     );
 
-    useEffect(() => setText(value), [value]);
+    useEffect(() => setText(value || defaultValue), [value]);
 
     const editHandler = useCallback(() => {
       setInputWidth(contentRef.current?.offsetWidth || 0);
       setIsEditing(true);
-    }, []);
+      if (text === defaultValue) {
+        setText('');
+      }
+    }, [text]);
 
     const blurHandler = useCallback(() => {
-      setValue?.((text || '').replace(/[\s\u00A0]+/g, ' ').trim());
+      const trimmedText = (text || '').replace(/[\s\u00A0]+/g, ' ').trim();
+      setValue?.(trimmedText || defaultValue);
       setIsEditing(false);
     }, [text, setValue]);
```

Comment:
You know what I'm going to say, already! :) `defaultValue` needs to be in dependencies array